### ISSUE TITLE
Create abstraction for storage.

### ISSUE THREAD

**Issue by wtrocki** [feedhenry/fh-sync-js#7]
Allow to provide custom storage. Currently only lawnchair storage is supported, but mobile can also benefit from different storage solutions like pouchdb etc.

**Comment by wtrocki** [comment]
@wei-lee WDYT? Is that lawchair seems to be no longer actively maintained.

**Comment by wei-lee** [comment]
Yes, I think the sync framework should define it's own storage interface, and provide a default implementation.  Then user can override the storage implementation in their own apps if they want.

**Comment by evanshortiss** [comment]
Would be great. Could be as simple as having someone supply something like this:

```js
const engine = {
  create: (key, data) => {
    // Maybe return a Promise if it is async, e.g write to pouchdb or cordova filesystem 
    localStorage.setItem(key, JSON.stringify(data))
  },
  read: (ket, data) => {
    return new Promise((resolve, reject) => {
      try {
        resolve(JSON.parse(localStorage.getItem(key)));
      } catch (e) {
        reject(e);
      }
  },
  // etc
}

sync.init({storage: engine})
```

**Comment by wtrocki** [comment]
@emilioicai - Required work was merged to master. You are no longer blocked on this

**Comment by wtrocki** [comment]
Implemented and verified as part of #15.
